src/components/VariablesTable.vue

```diff
@@ -0,0 +1,189 @@
+          <span />
+        </template>
+
+        <template #action="{ row }">
+          <div class="variables-table__action">
+            <VariableMenu :variable="row" size="xs" @delete="refresh" />
+          </div>
+        </template>
+
+        <template #empty-state>
+          <PEmptyResults>
+            <template #message>
+              {{ localization.info.noVariables }}
+            </template>
+            <template v-if="isCustomFilter" #actions>
+              <p-button size="sm" secondary @click="clear">
+                Clear Filters
+              </p-button>
+            </template>
+          </PEmptyResults>
+        </template>
+      </p-table>
+
+      <template #footer-end>
+        <p-pager v-if="variables.length" v-model:page="offset" :pages="variablesCount" />
+      </template>
+    </p-layout-table>
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { PTable, PEmptyResults, CheckboxModel } from '@prefecthq/prefect-design'
+  import { useDebouncedRef, useSubscription } from '@prefecthq/vue-compositions'
+  import { computed, ref } from 'vue'
+  import { FlowsDeleteButton, VariableMenu, ResultsCount, SearchInput, SelectedCount } from '@/components'
+  import { useCan, useVariablesFilter, useWorkspaceApi } from '@/compositions'
+  import { localization } from '@/localization'
+  import { VariablesFilter } from '@/models/Filters'
+  import { variableSortOptions } from '@/types'
+  import { formatDateTimeNumeric } from '@/utilities/dates'
+
+  const props = defineProps<{
+    filter?: VariablesFilter,
+  }>()
+
+  const api = useWorkspaceApi()
+  const can = useCan()
+
+  const variableLike = ref<string>()
+  const variableLikeDebounced = useDebouncedRef(variableLike, 1000)
+  const offset = ref(0)
+
+  const { filter, isCustomFilter, clear } = useVariablesFilter({
+    ...props.filter,
+    variables: {
+      ...props.filter?.variables,
+      nameLike: variableLikeDebounced,
+      valueLike: variableLikeDebounced,
+    },
+    offset,
+  })
+
+  const columns = [
+    {
+      label: 'selection',
+      width: '20px',
+      visible: can.delete.variable,
+    },
+    {
+      property: 'name',
+      label: 'Name',
+      width: '125px',
+    },
+    {
+      property: 'value',
+      label: 'Value',
+      width: '125px',
+    },
+    {
+      property: 'updated',
+      label: 'Updated',
+      width: '125px',
+    },
+    {
+      label: 'Action',
+      width: '42px',
+    },
+  ]
+
+  const selectedVariables = ref<string[]>([])
+  const selectAllVariables = (allVariablesSelected: CheckboxModel): string[] => {
+    if (allVariablesSelected) {
+      return selectedVariables.value = [...variables.value.map(variable => variable.id)]
+    }
+    return selectedVariables.value = []
+  }
+
+  const model = computed({
+    get() {
+      return selectedVariables.value.length === variables.value.length
+    },
+    set(value: boolean) {
+      selectAllVariables(value)
+    },
+  })
```

Comment:
We should make a select all component for this. Takes an array v-model and an array of all the possible values.

Reply:
Oh yea good thought, we're doing this in a few places. I'll open a ticket 👍🏻

Reply:
[Issue](https://github.com/PrefectHQ/prefect-ui-library/issues/1297)